weblab/entities/tests/test_views.py

```diff
@@ -84,11 +84,75 @@ def test_create_protocol_requires_permissions(self, logged_in_user, client):
         assert response.status_code == 302
         assert '/login/' in response.url
 
+@pytest.mark.django_db
+class TestEntityRenaming:
+    def test_model_renaming_success(self, client, logged_in_user, helpers):
+        helpers.add_permission(logged_in_user, 'create_model')
+        model = recipes.model.make(author=logged_in_user)
+        assert model.name == 'my model1'
+
+        response = client.post(
+            '/entities/models/%d/rename' % model.pk,
+            data={
+                     'name': 'new name'
+                 })
+        assert response.status_code == 302
+        entity = ModelEntity.objects.first()
+        assert entity.name == 'new name'
+
+    def test_model_renaming_failure(self, client, logged_in_user, helpers):
+        helpers.add_permission(logged_in_user, 'create_model')
+        model = recipes.model.make(author=logged_in_user)
+
+        model2 = recipes.model.make(name='test model 2')
+        assert model.name == 'my model1'
+        assert model2.name == 'test model 2'
+
+        response = client.post(
+            '/entities/models/%d/rename' % model.pk,
+            data={
+                     'name': 'test model 2'
+                 })
+        assert response.status_code == 200
+        entity = ModelEntity.objects.first()
+        assert entity.name == 'my model1'
+
+    def test_model_and_protocol_renaming_success(self, client, logged_in_user, helpers):
+        helpers.add_permission(logged_in_user, 'create_model')
+        model = recipes.model.make(author=logged_in_user)
+
+        protocol = recipes.protocol.make(name='test protocol')
+        assert model.name == 'my model1'
+
+        response = client.post(
+            '/entities/models/%d/rename' % model.pk,
+            data={
+                     'name': 'test protocol'
+                 })
+        assert response.status_code == 302
+        entity = ModelEntity.objects.first()
+        assert entity.name == 'test protocol'
+
+    def test_model_abs_path_the_same(self, client, logged_in_user, helpers):
+        helpers.add_permission(logged_in_user, 'create_model')
+        model = recipes.model.make(author=logged_in_user)
+        commit = helpers.add_version(model, visibility='private')
+
+        abs_path = model.repo_abs_path
+        response = client.post(
+            '/entities/models/%d/rename' % model.pk,
+            data={
+                     'name': 'test protocol'
+                 })
+        assert response.status_code == 302
+        assert abs_path == model.repo_abs_path
```

Comment:
`model` here won't actually reflect changes in the database. So instead you want to check `entity.repo_abs_path` after checking the name.

Reply:
thanks, change made